src/subprocess.rs

```diff
@@ -1,26 +1,104 @@
 use crate::config::{DEFAULT_ENV_NAME, ENV_VAR_PREFIX};
 use crate::environments::Environments;
+use crate::graphql::GraphQLError;
 use crate::parameters::Parameters;
+use crate::warn_user;
 use color_eyre::eyre::Result;
 use std::collections::HashMap;
-use std::env;
+use std::fmt::{self, Display, Formatter};
+use std::str::FromStr;
+use std::{env, error};
 use subprocess::Exec;
 
 // for improved readability
 pub type EnvSettings = HashMap<String, String>;
 
+// NOTE: Hash and Debug are needed for testing... sigh
+#[derive(PartialEq, Eq, Hash, Debug)]
+pub enum Inheritance {
+    None,
+    Underlay,
+    Overlay,
+    Exclusive,
+}
+
+impl Display for Inheritance {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        match self {
+            Inheritance::None => write!(f, "none"),
```

Comment:
Since all you're doing is downcasing the value, you could do what we do for `Resource`:

```rust
write!(f, "{}", format!("{:?}", self).to_lowercase())
```

That would cover all your cases. I suppose it could break if someone overrode the `Debug` implementation, but that seems very unlikely to me. And even then, a test would likely catch it.

Reply:
Great call. Earlier, I had a different value that needed special handling, but this simplifies things.